lib/ddtrace/distributed.rb

```diff
@@ -0,0 +1,38 @@
+require 'ddtrace/span'
+
+module Datadog
+  # Common code related to distributed tracing.
+  module Distributed
+    module_function
+
+    # Parses a trace_id and a parent_id, typically sent as headers in
+    # a distributed tracing context, and returns a couple of trace_id,parent_id
+    # which are garanteed to be both non-zero. This does not 100% ensure they
+    # are valid (after all, the caller could mess up data) but at least it
+    # sorts out most common errors, such as syntax, nil values, etc.
+    # Both headers must be set, else nil values are returned, for both.
+    # Reports problem on debug log.
+    def parse_trace_headers(trace_id_header, parent_id_header)
```

Comment:
That's perfect, so we're having the right protection in the good place! thank you a lot!